File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.py
def solution(a, b, c, d):
    answer = 0
    count = {}
    for i in [a,b,c,d]:
        try: count[i] += 1
        except: count[i] = 1
    convert = {m:n for n,m in count.items()}
    du = [] #중복 원소 리스트
    du_x = [] #중복 아닌 원소 리스트
    for n,m in count.items():
        if m>=2: du.append(n)
        if m<2: du_x.append(n)
        if len(du_x)==0: answer=1111*n
        if len(du)==1 and len(du_x)==1: answer=(10*du[0]+du_x[0])**2
        if len(du)==2 and len(du_x)==0: answer=(du[0]+du[1])*abs(du[0]-du[1])
        if len(du)==1 and len(du_x)==2: answer=du_x[0]*du_x[1]
        if len(du)==0: answer=min(a,b,c,d)
    return answer

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.java
import java.util.Arrays;

class Solution {
    public int solution(int a, int b, int c, int d) {
        int answer = 0;
        int array[] = {a,b,c,d};
        Arrays.sort(array);
        
        if(array[0]!=array[1] && array[1]!=array[2] && array[2]!=array[3]){
            answer = array[0];
        }else if(array[2]!=array[3] && array[0]==array[1] && array[0]==array[2]){       //[1,1,1,2]
            answer = (10*array[0]+array[3])*(10*array[0]+array[3]);                   
        }else if(array[0]!=array[1] && array[2]==array[3] && array[1]==array[2]){       //[1,2,2,2]
            answer = (10*array[3]+array[0])*(10*array[3]+array[0]);
        }else if(array[1]!=array[2] && array[0]==array[1] && array[2]==array[3]){       //[1,1,2,2]
            answer = (array[0]+array[3])*Math.abs(array[0]-array[3]);
        }else if(array[1]!=array[2] && array[0]!=array[1] && array[2]==array[3]){       //[1,2,3,3]   
            answer = array[0]*array[1];
        }else if(array[0]==array[1] && array[1]!=array[2] && array[2]!=array[3]){       //[1,1,2,3]
            answer = array[2]*array[3];
        }else if(array[0]!=array[1] && array[2]!=array[3] && array[1]==array[2]){       //[1,2,2,3]
            answer = array[0]*array[3];
        }else if(array[0]==array[1] && array[1]==array[2] && array[2]==array[3]){       //[1,1,1,1]
            answer = array[0]*1111;
        }else{                                                                          //[1,2,3,4]
            answer = array[0];
        }
        return answer;
    }
}